src/librustc_typeck/diagnostics.rs

```diff
@@ -1130,6 +1244,80 @@ impl Bytes { ... } // error, same as above
 ```
 "##,
 
+E0117: r##"
+You tried to implement a trait on a type which isn't defined in your crate.
```

Comment:
s/on/for/.

Also this explanation is a bit too specialised. This error will happen iff trait, trait’s type parameters and type the trait is implemented for are all foreign (i.e. not from your crate). That being said, there’s at least 3 possible fixes: define a newtype in your crate, change trait’s type parameter to something from your crate or implement a trait declared in your crate instead (e.g. declaring `trait MyCopy : Copy {}` or something along the lines and implementing that instead).